### PR TITLE
Edit study

### PR DESCRIPTION
**Pull request by Okanmercan99**

### CODE REVIEW

File: src/main/java/io/passport/server/service/ExperimentService.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        experimentRepository.saveAll(ExperimentEntries);
+    }
+
+    /**
+     * Delete all Experiment entries related to the study.
+     * @param studyId ID of the study
+     * @return
+     */
+    @Transactional
+    public void clearExperimentEntriesByStudyId(Long studyId) {
```
Comment (dogukan10) [inline src/main/java/io/passport/server/service/ExperimentService.java:65]: Let's make it private.

File: src/main/java/io/passport/server/service/StudyPersonnelService.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        return this.studyPersonnelRepository.findPersonnelByStudyId(studyId);
+    }
+
+    /**
+     * Delete all StudyPersonnel entries related to the study.
+     * @param studyId ID of the study
+     * @return
+     */
+    @Transactional
+    public void clearStudyPersonnelEntriesByStudyId(Long studyId) {
```
Comment (dogukan10) [inline src/main/java/io/passport/server/service/StudyPersonnelService.java:45]: Let's make it private.

File: src/main/java/io/passport/server/controller/ExperimentController.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+     * Clear all old Experiment entries related to the study and create new ones. Return updated experiment list.
+     * @param studyId ID of the study.
+     * @param experiments List of experiment to be used in experiment entries
+     * @return
+     */
+    @PostMapping()
+    public ResponseEntity<?> createExperiments(@RequestParam Long studyId, @RequestBody List<Experiment> experiments) {
+        try{
+            this.experimentService.createExperimentEntries(studyId, experiments);
+            List<Experiment> newExperiments = this.experimentService.findExperimentByStudyId(studyId);
```
Comment (dogukan10) [inline src/main/java/io/passport/server/controller/ExperimentController.java:56]: Why do we fetch the new experiments ? We can modify experimentService.createExperimentEntries to return the new ones right ? I think saveAll returns the new list. Can you please check ? If you update this code, please use this logic in other controllers as well.

**Review by dogukan10 — CHANGES_REQUESTED** [review]

File: src/main/java/io/passport/server/service/ExperimentService.java (hunk 1 above)
Comment (Okanmercan99) [reply src/main/java/io/passport/server/service/ExperimentService.java:65]: Transactional annotation cannot be used with private methods.

File: src/main/java/io/passport/server/service/StudyPersonnelService.java (hunk 2 above)
Comment (Okanmercan99) [reply src/main/java/io/passport/server/service/StudyPersonnelService.java:45]: Transactional annotation cannot be used with private methods.

**Review by dogukan10 — APPROVED** [review]

**Pull request merged**